xlgui/devices.py

```diff
@@ -33,62 +33,54 @@
 from xl.nls import gettext as _
 from xl import xdg, settings, event, devices
 from xlgui import collection
+from xlgui.guiutil import GtkTemplate
 
 logger = logging.getLogger(__name__)
 
 
-
-
-class ManagerDialog(object):
+@GtkTemplate('ui', 'device_manager.ui')
+class ManagerDialog(Gtk.Window):
     """
         the device manager dialog
     """
 
+    __gtype_name__ = 'DeviceManager'
+
+    btn_add, btn_edit, btn_remove, tree_devices = GtkTemplate.Child.widgets(4)
+
     def __init__(self, parent, main):
+        Gtk.Window.__init__(self)
+        self.init_template()
+
         self.main = main
-        self.parent = parent
         self.device_manager = self.main.exaile.devices
-        self.builder = Gtk.Builder()
-        self.builder.add_from_file(xdg.get_data_path('ui', 'device_manager.ui'))
-        self.window = self.builder.get_object('device_manager')
-        self.window.set_transient_for(self.parent)
-        self.window.set_position(Gtk.WindowPosition.CENTER_ON_PARENT)
-        self.window.connect('delete-event', self.on_close)
-
-        self.builder.connect_signals({
-            'on_btn_connect_clicked': self.on_connect,
-            'on_btn_disconnect_clicked': self.on_disconnect,
-            'on_btn_edit_clicked': self.on_edit,
-            'on_btn_add_clicked': self.on_add,
-            'on_btn_remove_clicked': self.on_remove,
-            'on_btn_close_clicked': self.on_close,
-            })
+        self.set_transient_for(parent)
 
         # TODO: make these actually work.  For now, they are hidden
-        for item in ('add', 'edit', 'remove'):
-            self.builder.get_object('btn_%s' % item).destroy()
+        self.btn_add.destroy()
+        self.btn_edit.destroy()
+        self.btn_remove.destroy()
 
         # object should really be devices.Device, but it doesnt work :/
         self.model = Gtk.ListStore(object, GdkPixbuf.Pixbuf, str, str)
-        self.tree = self.builder.get_object('tree_devices')
-        self.tree.set_model(self.model)
+        self.tree_devices.set_model(self.model)
 
         render = Gtk.CellRendererPixbuf()
```

Comment:
That would be a great testing tool.... though I don't actually like having the device manager, really, as Exaile has such poor device support I think I'd almost rather get rid of it totally. I'm not opinionated enough about it yet to actually do that, however.

However, it looks like enabling the CD plugin and inserting an audio disc will put something in the device manager. Most people tend to have CD's somewhere...